Mathlib/Init/Data/Fin/Basic.lean

```diff
@@ -9,18 +9,14 @@ import Mathlib.Init.Data.Nat.Notation
 # Theorems about equality in `Fin`.
 -/
 
-set_option autoImplicit true
-
 namespace Fin
 
-theorem eq_of_veq : ∀ {i j : Fin n}, i.val = j.val → i = j
-  | ⟨iv, ilt₁⟩, ⟨jv, jlt₁⟩, h => by cases h; rfl
-
-theorem veq_of_eq : ∀ {i j : Fin n}, i = j → i.val = j.val
-  | ⟨_, _⟩, _, rfl => rfl
+variable {n : ℕ} {i j : Fin n}
 
-theorem ne_of_vne {i j : Fin n} (h : i.val ≠ j.val) : i ≠ j := fun h' ↦ absurd (veq_of_eq h') h
+@[deprecated] theorem eq_of_veq : i.val = j.val → i = j := eq_of_val_eq -- 2024-02-15
+@[deprecated] theorem veq_of_eq : i = j → i.val = j.val := val_eq_of_eq -- 2024-02-15
```

Comment:
Can you use `alias`? I had the vague impression that that was better for some reason. Otherwise I think you should put the new name in the attribute as well.

Reply:
`alias` is not available in this file and I don't want to add `import`s to `Mathlib/Init/*`. Added names to `@[deprecated]`.